src/signer.rs

```diff
@@ -0,0 +1,68 @@
+use core::marker::PhantomData;
+
+use hashbrown::HashMap;
+use move_core_types::account_address::AccountAddress;
+
+use crate::{balance::BalanceOf, Config, Error, Pallet};
+
+/// A simple signature.
+#[derive(PartialEq, Eq)]
+enum Signature {
+    /// Signer has approved the script with the signature.
+    Approved,
+    /// Signer's approval still missing.
+    Missing,
+}
+
+/// Script signature handler tracks required signatures for the provided script.
+pub(crate) struct ScriptSignatureHandler<T: Config>
+where
+    BalanceOf<T>: From<u128> + Into<u128>,
+{
+    _pd_config: PhantomData<T>,
+
+    /// All required signers.
+    signers: HashMap<AccountAddress, Signature>,
+}
+
+impl<T: Config> ScriptSignatureHandler<T>
+where
+    BalanceOf<T>: From<u128> + Into<u128>,
+{
+    /// Creates a new [`ScriptSignatureHandler`] with all blank signatures for the provided script.
+    pub(crate) fn new(script_args: &[Vec<u8>], signer_count: usize) -> Result<Self, Error<T>> {
+        if signer_count > script_args.len() {
+            return Err(Error::ScriptSignatureFailure);
+        }
+
+        let mut signers = HashMap::new();
+        for signer in &script_args[..signer_count] {
+            let account_address =
+                bcs::from_bytes(signer).map_err(|_| Error::UnableToDeserializeAccount)?;
+            signers.insert(account_address, Signature::Missing);
+        }
+
+        Ok(Self {
+            _pd_config: PhantomData,
+            signers,
+        })
+    }
+
+    /// Sign the script.
+    pub(crate) fn sign_script(&mut self, account: &T::AccountId) -> Result<(), Error<T>> {
+        let account_address = Pallet::<T>::to_move_address(account)?;
+
+        if let Some(signature) = self.signers.get_mut(&account_address) {
```

Comment:
Shouldn't we return any error if someone is trying to sign a script when he is not in a signer list?
Because now it seems like we're silently ignore the fact that someone is trying to sign something what was not addressed to him.

Like you know, I call it script signing which is returning me Ok() even if it signed nothing 😄

Reply:
I understood that we're not returning any errors here to handle scripts without any signer parameters properly, as those with signer (which is not found here) will be catched by the next statement in the main execute function, but maybe a little comment would make it easier to understand without tracking the execute function.

Reply:
It's like when you try to fake your mother's signature in school - you can put it there, but no one will care since that's not her genuine signature 😃

Reply:
> but maybe a little comment would make it easier to understand without tracking the execute function.

And after some extra discussion on Slack 💪 
Done